homeassistant/components/denonavr/media_player.py

```diff
@@ -249,9 +249,29 @@ def __init__(
             and MediaPlayerEntityFeature.SELECT_SOUND_MODE
         )
 
+        self._receiver.register_callback("ALL", self._telnet_callback)
```

Comment:
It would have been a nicer design if this returned a callable that canceled the registration since we could than wrap it in `self.async_on_remove`

Reply:
Not sure I understand this proposal but it seems there are components that follow the model I have here. This also sounds like it requires a change to the underlying library?

It seems like this is an acceptable pattern based on
https://github.com/home-assistant/core/blob/dev/homeassistant/components/nobo_hub/sensor.py#L77 as one example.

Reply:
> Not sure I understand this proposal but it seems there are components that follow the model I have here. This also sounds like it requires a change to the underlying library?

Yes it would

You could do `self.async_on_remove(self._receiver.register_callback("ALL", self._telnet_callback))` if `register_callback` returned a callable that canceled the registration.

> 
> It seems like this is an acceptable pattern based on [`dev`/homeassistant/components/nobo_hub/sensor.py#L77](https://github.com/home-assistant/core/blob/dev/homeassistant/components/nobo_hub/sensor.py?rgh-link-date=2023-01-23T01%3A55%3A32Z#L77) as one example.

Its a nice to have for sure.

Reply:
Ah cool, well thanks for explaining. This is at least good advice for another integration I'm working on!